### PR TITLE
Implement `ClaimStakeReward` action

### PR DESCRIPTION
**Pull request by moreal**
This pull request continues from #952.

### CODE REVIEW

File: Lib9c/Action/ClaimStakeReward.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        public ClaimStakeReward() : base()
+        {
+        }
+
+        public override IAccountStateDelta Execute(IActionContext context)
+        {
+            var states = context.PreviousStates;
+            if (!states.TryGetStakeState(context.Signer, out StakeState stakeState))
+            {
+                throw new InvalidOperationException();
```
Comment (ipdae) [inline Lib9c/Action/ClaimStakeReward.cs:31]: we already implemented `FailedLoadStateException` this case.

File: Lib9c/Action/ClaimStakeReward.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+            var currency = states.GetGoldCurrency();
+            var balance = states.GetBalance(context.Signer, currency);
+
+            if (!stakeState.IsClaimable(context.BlockIndex))
+            {
+                throw new RequiredBlockIndexException();
+            }
+
+            var avatarState = states.GetAvatarState(AvatarAddress);
```
Comment (ipdae) [inline Lib9c/Action/ClaimStakeReward.cs:45]:
```suggestion
            var avatarState = states.GetAvatarStateV2(AvatarAddress);
```

**Review by ipdae — COMMENTED** [review]
it would be better some integration test case for staking.

File: Lib9c/Action/ClaimStakeReward.cs (hunk 2 above)
Comment (moreal) [reply Lib9c/Action/ClaimStakeReward.cs]: I applied this suggestion with [95d1364](https://github.com/planetarium/lib9c/pull/962/commits/95d1364a97ca3eaf4240ff8f8647fcc1bbd523cc).

File: Lib9c/Action/ClaimStakeReward.cs (hunk 1 above)
Comment (moreal) [reply Lib9c/Action/ClaimStakeReward.cs]: I applied it with [3f73165](https://github.com/planetarium/lib9c/pull/962/commits/3f7316537df86363ac79d1ffc901776097dfb5c8)

File: Lib9c/Action/ClaimStakeReward.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        public override IAccountStateDelta Execute(IActionContext context)
+        {
+            var states = context.PreviousStates;
+            if (!states.TryGetStakeState(context.Signer, out StakeState stakeState))
+            {
+                throw new InvalidOperationException();
+            }
+
+            var stakeRegularRewardSheet = states.GetSheet<StakeRegularRewardSheet>();
+            var stakeAchievementRewardSheet = states.GetSheet<StakeAchievementRewardSheet>();
```
[the remark this answers is not in the document]
Comment (moreal) [reply Lib9c/Action/ClaimStakeReward.cs]: I applied it with [a3522b6](https://github.com/planetarium/lib9c/pull/962/commits/a3522b644933a5a66db1702aa26d6ad491100dbb)

**Comment by ipdae** [comment]
test failed. 😓

**Comment by moreal** [comment]
> test failed. 😓

Thanks to let know it. It was my mistake.. I fixed it 😥

**Review by ipdae — APPROVED** [review]

**Pull request merged**